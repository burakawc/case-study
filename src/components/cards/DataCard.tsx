import React from 'react';
import { Card, Typography, Tag, Image, Avatar, Space, Badge } from 'antd';
import { UserOutlined } from '@ant-design/icons';
import ActionButtons from '../ActionButtons';
import type { Product, User } from '@/types';

const { Title } = Typography;

interface DataCardProps {
  data: Product | User;
  type: 'product' | 'user';
  onView?: () => void;
  onEdit?: () => void;
  onDelete?: () => void;
  onToggleFavorite?: () => void;
  isFavorite?: boolean;
  showFavorite?: boolean;
}

/**
 * Reusable data card component for displaying products or users
 * 
 * @param data - Product or User data to display
 * @param type - Type of data ('product' or 'user')
 * @param onView - Function called when view button is clicked
 * @param onEdit - Function called when edit button is clicked
 * @param onDelete - Function called when delete button is clicked
 * @param onToggleFavorite - Function called when favorite button is clicked
 * @param isFavorite - Whether the item is currently favorited
 * @param showFavorite - Whether to show the favorite button
 * @returns JSX element containing the data card
 */
const DataCard: React.FC<DataCardProps> = ({
  data,
  type,
  onView,
  onEdit,
  onDelete,
  onToggleFavorite,
  isFavorite = false,
  showFavorite = false
}) => {
  const isProduct = type === 'product';
  const product = isProduct ? data as Product : null;
  const user = !isProduct ? data as User : null;

  const renderProductContent = () => {
    if (!product) return null;
    
    return (
      <>
        <div style={{ display: 'flex', alignItems: 'flex-start', marginBottom: 12 }}>
          <div style={{
            width: 56,
            height: 56,
            minWidth: 56,
            minHeight: 56,
            display: 'flex',
            alignItems: 'center',
            justifyContent: 'center',
            borderRadius: 12,
            marginRight: 14,
            overflow: 'hidden',
          }}>
            <Image
              src={product.thumbnail}
              alt={product.title}
              width={48}
              height={48}
              style={{ objectFit: 'cover', borderRadius: 8, background: 'transparent' }}
              preview={false}
            />
          </div>
          <div style={{ flex: 1, minWidth: 0 }}>
            <Title level={5} style={{ margin: 0, marginBottom: 4, wordBreak: 'break-word' }}>
              {product.title}
            </Title>
            <div style={{ fontSize: '12px', color: '#666', marginBottom: 8, wordBreak: 'break-word' }}>
              {product.description.length > 50 
                ? `${product.description.substring(0, 50)}...` 
                : product.description
              }
            </div>
            <Space size="small" wrap>
              <Tag color="blue">{product.brand}</Tag>
              <Tag color="green">{product.category}</Tag>
            </Space>
          </div>
        </div>
        
        <div style={{ display: 'flex', justifyContent: 'space-between', alignItems: 'center' }}>
          <div>
            <div style={{ fontSize: '18px', fontWeight: 'bold', color: '#1890ff' }}>
              ${product.price}
            </div>
            <div style={{ fontSize: '12px', color: '#666' }}>
              Stock: {product.stock}
            </div>
          </div>
          <div style={{ textAlign: 'right' }}>
            <Tag color={product.rating >= 4 ? 'green' : product.rating >= 3 ? 'orange' : 'red'}>
              {product.rating.toFixed(1)} ‚≠ê
            </Tag>
          </div>
        </div>
      </>
    );
  };

  const renderUserContent = () => {
    if (!user) return null;
    
    return (
      <>
        <div style={{ display: 'flex', alignItems: 'center', marginBottom: 12 }}>
          <Avatar 
            src={user.image} 
            size={60}
            icon={<UserOutlined />}
            style={{ marginRight: 12 }}
          />
          <div style={{ flex: 1 }}>
            <Title level={5} style={{ margin: 0, marginBottom: 4 }}>
              {user.firstName} {user.lastName}
            </Title>
            <div style={{ fontSize: '12px', color: '#666', marginBottom: 8 }}>
              @{user.username}
            </div>
            <div style={{ fontSize: '12px', color: '#666', marginBottom: 8 }}>
              {user.email}
            </div>
            <Space size="small">
              <Tag color={user.gender === 'male' ? 'blue' : user.gender === 'female' ? 'pink' : 'default'}>
                {user.gender}
              </Tag>
              <Tag color="red">{user.bloodGroup}</Tag>
            </Space>
          </div>
        </div>
        
        <div style={{ display: 'flex', justifyContent: 'space-between', alignItems: 'center' }}>
          <div>
            <div style={{ fontSize: '14px', color: '#666' }}>
              Age: {user.age}
            </div>
            <div style={{ fontSize: '12px', color: '#666' }}>
              {user.company.name}
            </div>
          </div>
          <div style={{ textAlign: 'right' }}>
            <Badge count={user.age} style={{ backgroundColor: '#1890ff' }} />
          </div>
        </div>
      </>
    );
  };

  return (
    <Card
      hoverable
      style={{ marginBottom: 16 }}
      bodyStyle={{ padding: 16 }}
      actions={[
        <ActionButtons
          key="actions"
          onView={onView}
          onEdit={onEdit}
          onDelete={onDelete}
          onToggleFavorite={onToggleFavorite}
          isFavorite={isFavorite}
          showFavorite={showFavorite}
          deleteConfirmTitle={`Are you sure you want to delete this ${type}?`}
        />
      ]}
    >
      {isProduct ? renderProductContent() : renderUserContent()}
    </Card>
  );
};

export default DataCard; 