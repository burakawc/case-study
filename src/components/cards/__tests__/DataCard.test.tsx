import { render, screen, fireEvent } from '@testing-library/react';
import { describe, it, expect, vi } from 'vitest';
import DataCard from '../DataCard';
import type { Product, User } from '@/types';

// Mock ActionButtons component
vi.mock('../../ActionButtons', () => ({
  default: ({ onView, onEdit, onDelete, onToggleFavorite, isFavorite, showFavorite }: any) => (
    <div data-testid="action-buttons">
      {onView && <button onClick={onView} data-testid="view-btn">View</button>}
      {onEdit && <button onClick={onEdit} data-testid="edit-btn">Edit</button>}
      {onDelete && <button onClick={onDelete} data-testid="delete-btn">Delete</button>}
      {showFavorite && (
        <button onClick={onToggleFavorite} data-testid="favorite-btn">
          {isFavorite ? '‚ù§Ô∏è' : 'ü§ç'}
        </button>
      )}
    </div>
  ),
}));

describe('DataCard', () => {
  const mockProduct: Product = {
    id: 1,
    title: 'Test Product',
    description: 'This is a test product description that is longer than 50 characters to test truncation',
    price: 99.99,
    discountPercentage: 10,
    rating: 4.5,
    stock: 100,
    brand: 'TestBrand',
    category: 'Electronics',
    thumbnail: 'https://example.com/thumbnail.jpg',
    images: ['https://example.com/image1.jpg', 'https://example.com/image2.jpg'],
    createdAt: '2023-01-01T00:00:00.000Z',
    updatedAt: '2023-01-01T00:00:00.000Z',
  };

  const mockUser: User = {
    id: 1,
    firstName: 'John',
    lastName: 'Doe',
    age: 30,
    gender: 'male',
    email: 'john.doe@example.com',
    phone: '+1234567890',
    username: 'johndoe',
    password: 'password123',
    birthDate: '1993-01-01',
    image: 'https://example.com/avatar.jpg',
    bloodGroup: 'A+',
    height: 180,
    weight: 75,
    eyeColor: 'brown',
    hair: {
      color: 'black',
      type: 'straight',
    },
    domain: 'example.com',
    ip: '192.168.1.1',
    address: {
      address: '123 Main St',
      city: 'New York',
      coordinates: {
        lat: 40.7128,
        lng: -74.0060,
      },
      postalCode: '10001',
      state: 'NY',
    },
    macAddress: '00:11:22:33:44:55',
    university: 'Test University',
    bank: {
      cardExpire: '12/25',
      cardNumber: '1234567890123456',
      cardType: 'visa',
      currency: 'USD',
      iban: 'US12345678901234567890',
    },
    company: {
      address: {
        address: '456 Business Ave',
        city: 'New York',
        coordinates: {
          lat: 40.7128,
          lng: -74.0060,
        },
        postalCode: '10002',
        state: 'NY',
      },
      department: 'Engineering',
      name: 'Test Company',
      title: 'Software Engineer',
    },
    ein: '12-3456789',
    ssn: '123-45-6789',
    userAgent: 'Mozilla/5.0...',
    createdAt: '2023-01-01T00:00:00.000Z',
    updatedAt: '2023-01-01T00:00:00.000Z',
  };

  describe('Product Card', () => {
    it('should render product card correctly', () => {
      const { container } = render(
        <DataCard
          data={mockProduct}
          type="product"
          onView={vi.fn()}
          onEdit={vi.fn()}
          onDelete={vi.fn()}
        />
      );

      expect(screen.getByText('Test Product')).toBeInTheDocument();
      expect(screen.getByText((content) =>
        content.startsWith('This is a test product description that is longer')
      )).toBeInTheDocument();
      expect(screen.getByText('TestBrand')).toBeInTheDocument();
      expect(screen.getByText('Electronics')).toBeInTheDocument();
      expect(screen.getByText('$99.99')).toBeInTheDocument();
      expect(screen.getByText('Stock: 100')).toBeInTheDocument();
      expect(screen.getByText('4.5 ‚≠ê')).toBeInTheDocument();
      expect(screen.getByTestId('action-buttons')).toBeInTheDocument();

      // Snapshot test
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render product card with favorite functionality', () => {
      const onToggleFavorite = vi.fn();
      const { container } = render(
        <DataCard
          data={mockProduct}
          type="product"
          onView={vi.fn()}
          onEdit={vi.fn()}
          onDelete={vi.fn()}
          onToggleFavorite={onToggleFavorite}
          isFavorite={true}
          showFavorite={true}
        />
      );

      const favoriteButton = screen.getByTestId('favorite-btn');
      expect(favoriteButton).toBeInTheDocument();
      expect(favoriteButton).toHaveTextContent('‚ù§Ô∏è');

      fireEvent.click(favoriteButton);
      expect(onToggleFavorite).toHaveBeenCalledTimes(1);

      // Snapshot test
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should handle long product descriptions correctly', () => {
      const longDescriptionProduct = {
        ...mockProduct,
        description: 'This is a very long product description that should be truncated when it exceeds 50 characters in length',
      };

      render(
        <DataCard
          data={longDescriptionProduct}
          type="product"
          onView={vi.fn()}
          onEdit={vi.fn()}
          onDelete={vi.fn()}
        />
      );

      expect(screen.getByText((content) =>
        content.startsWith('This is a very long product description that shoul')
      )).toBeInTheDocument();
    });

    it('should render different rating colors based on rating value', () => {
      const lowRatingProduct = { ...mockProduct, rating: 2.5 };
      const mediumRatingProduct = { ...mockProduct, rating: 3.5 };
      const highRatingProduct = { ...mockProduct, rating: 4.5 };

      const { rerender } = render(
        <DataCard
          data={lowRatingProduct}
          type="product"
          onView={vi.fn()}
          onEdit={vi.fn()}
          onDelete={vi.fn()}
        />
      );

      expect(screen.getByText('2.5 ‚≠ê')).toBeInTheDocument();

      rerender(
        <DataCard
          data={mediumRatingProduct}
          type="product"
          onView={vi.fn()}
          onEdit={vi.fn()}
          onDelete={vi.fn()}
        />
      );

      expect(screen.getByText('3.5 ‚≠ê')).toBeInTheDocument();

      rerender(
        <DataCard
          data={highRatingProduct}
          type="product"
          onView={vi.fn()}
          onEdit={vi.fn()}
          onDelete={vi.fn()}
        />
      );

      expect(screen.getByText('4.5 ‚≠ê')).toBeInTheDocument();
    });
  });

  describe('User Card', () => {
    it('should render user card correctly', () => {
      const { container } = render(
        <DataCard
          data={mockUser}
          type="user"
          onView={vi.fn()}
          onEdit={vi.fn()}
          onDelete={vi.fn()}
        />
      );

      expect(screen.getByText('John Doe')).toBeInTheDocument();
      expect(screen.getByText('@johndoe')).toBeInTheDocument();
      expect(screen.getByText('john.doe@example.com')).toBeInTheDocument();
      expect(screen.getByText('male')).toBeInTheDocument();
      expect(screen.getByText('A+')).toBeInTheDocument();
      expect(screen.getByText('Age: 30')).toBeInTheDocument();
      expect(screen.getByText('Test Company')).toBeInTheDocument();

      // Snapshot test
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render user card with different gender colors', () => {
      const femaleUser = { ...mockUser, gender: 'female' as const };
      const otherUser = { ...mockUser, gender: 'other' as const };

      const { rerender } = render(
        <DataCard
          data={femaleUser}
          type="user"
          onView={vi.fn()}
          onEdit={vi.fn()}
          onDelete={vi.fn()}
        />
      );

      expect(screen.getByText('female')).toBeInTheDocument();

      rerender(
        <DataCard
          data={otherUser}
          type="user"
          onView={vi.fn()}
          onEdit={vi.fn()}
          onDelete={vi.fn()}
        />
      );

      expect(screen.getByText('other')).toBeInTheDocument();
    });

    it('should render user card with favorite functionality', () => {
      const onToggleFavorite = vi.fn();
      const { container } = render(
        <DataCard
          data={mockUser}
          type="user"
          onView={vi.fn()}
          onEdit={vi.fn()}
          onDelete={vi.fn()}
          onToggleFavorite={onToggleFavorite}
          isFavorite={false}
          showFavorite={true}
        />
      );

      const favoriteButton = screen.getByTestId('favorite-btn');
      expect(favoriteButton).toBeInTheDocument();
      expect(favoriteButton).toHaveTextContent('ü§ç');

      fireEvent.click(favoriteButton);
      expect(onToggleFavorite).toHaveBeenCalledTimes(1);

      // Snapshot test
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('Action Buttons', () => {
    it('should call action handlers when buttons are clicked', () => {
      const onView = vi.fn();
      const onEdit = vi.fn();
      const onDelete = vi.fn();

      render(
        <DataCard
          data={mockProduct}
          type="product"
          onView={onView}
          onEdit={onEdit}
          onDelete={onDelete}
        />
      );

      fireEvent.click(screen.getByTestId('view-btn'));
      expect(onView).toHaveBeenCalledTimes(1);

      fireEvent.click(screen.getByTestId('edit-btn'));
      expect(onEdit).toHaveBeenCalledTimes(1);

      fireEvent.click(screen.getByTestId('delete-btn'));
      expect(onDelete).toHaveBeenCalledTimes(1);
    });

    it('should not render action buttons when handlers are not provided', () => {
      render(
        <DataCard
          data={mockProduct}
          type="product"
        />
      );

      expect(screen.queryByTestId('view-btn')).not.toBeInTheDocument();
      expect(screen.queryByTestId('edit-btn')).not.toBeInTheDocument();
      expect(screen.queryByTestId('delete-btn')).not.toBeInTheDocument();
    });
  });

  describe('Edge Cases', () => {
    it('should handle missing optional props gracefully', () => {
      const { container } = render(
        <DataCard
          data={mockProduct}
          type="product"
        />
      );

      expect(container.firstChild).toBeInTheDocument();
      expect(screen.getByText('Test Product')).toBeInTheDocument();
    });

    it('should handle empty or null data gracefully', () => {
      const emptyProduct = {
        ...mockProduct,
        title: '',
        description: '',
        brand: '',
        category: '',
      };

      render(
        <DataCard
          data={emptyProduct}
          type="product"
        />
      );

      const headings = screen.getAllByRole('heading');
      headings.forEach(h => expect(h.textContent).toBe(''));
    });

    it('should not render user content if user is null', () => {
      const { container } = render(
        <DataCard
          // @ts-ignore
          data={null}
          type="user"
        />
      );
      // User i√ßeriƒüi olmamalƒ±
      expect(container.textContent).toBe('');
    });

    it('should not render product content if product is null', () => {
      // @ts-ignore
      const { container } = render(
        <DataCard
          // @ts-ignore
          data={null}
          type="product"
        />
      );
      // Product i√ßeriƒüi olmamalƒ±
      expect(container.textContent).toBe('');
    });
  });
}); 