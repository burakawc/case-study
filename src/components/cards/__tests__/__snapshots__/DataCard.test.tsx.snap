// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DataCard > Product Card > should render product card correctly 1`] = `
<div
  class="ant-card ant-card-bordered ant-card-hoverable css-dev-only-do-not-override-pjilya"
  style="margin-bottom: 16px;"
>
  <div
    class="ant-card-body"
    style="padding: 16px;"
  >
    <div
      style="display: flex; align-items: flex-start; margin-bottom: 12px;"
    >
      <div
        style="width: 56px; height: 56px; min-width: 56px; min-height: 56px; display: flex; align-items: center; justify-content: center; border-radius: 12px; margin-right: 14px; overflow: hidden;"
      >
        <div
          class="ant-image css-dev-only-do-not-override-pjilya"
          style="width: 48px; height: 48px;"
        >
          <img
            alt="Test Product"
            class="ant-image-img css-dev-only-do-not-override-pjilya"
            height="48"
            src="https://example.com/thumbnail.jpg"
            style="height: 48px; object-fit: cover; border-radius: 8px; background: transparent;"
            width="48"
          />
        </div>
      </div>
      <div
        style="flex: 1 1 0%; min-width: 0;"
      >
        <h5
          class="ant-typography css-dev-only-do-not-override-pjilya"
          style="margin: 0px 0px 4px 0px; word-break: break-word;"
        >
          Test Product
        </h5>
        <div
          style="font-size: 12px; color: rgb(102, 102, 102); margin-bottom: 8px; word-break: break-word;"
        >
          This is a test product description that is longer ...
        </div>
        <div
          class="ant-space css-dev-only-do-not-override-pjilya ant-space-horizontal ant-space-align-center ant-space-gap-row-small ant-space-gap-col-small"
          style="flex-wrap: wrap;"
        >
          <div
            class="ant-space-item"
          >
            <span
              class="ant-tag ant-tag-blue css-dev-only-do-not-override-pjilya"
            >
              TestBrand
            </span>
          </div>
          <div
            class="ant-space-item"
          >
            <span
              class="ant-tag ant-tag-green css-dev-only-do-not-override-pjilya"
            >
              Electronics
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      style="display: flex; justify-content: space-between; align-items: center;"
    >
      <div>
        <div
          style="font-size: 18px; font-weight: bold; color: rgb(24, 144, 255);"
        >
          $
          99.99
        </div>
        <div
          style="font-size: 12px; color: rgb(102, 102, 102);"
        >
          Stock: 
          100
        </div>
      </div>
      <div
        style="text-align: right;"
      >
        <span
          class="ant-tag ant-tag-green css-dev-only-do-not-override-pjilya"
        >
          4.5
           ⭐
        </span>
      </div>
    </div>
  </div>
  <ul
    class="ant-card-actions"
  >
    <li
      style="width: 100%;"
    >
      <span>
        <div
          data-testid="action-buttons"
        >
          <button
            data-testid="view-btn"
          >
            View
          </button>
          <button
            data-testid="edit-btn"
          >
            Edit
          </button>
          <button
            data-testid="delete-btn"
          >
            Delete
          </button>
        </div>
      </span>
    </li>
  </ul>
</div>
`;

exports[`DataCard > Product Card > should render product card with favorite functionality 1`] = `
<div
  class="ant-card ant-card-bordered ant-card-hoverable css-dev-only-do-not-override-pjilya"
  style="margin-bottom: 16px;"
>
  <div
    class="ant-card-body"
    style="padding: 16px;"
  >
    <div
      style="display: flex; align-items: flex-start; margin-bottom: 12px;"
    >
      <div
        style="width: 56px; height: 56px; min-width: 56px; min-height: 56px; display: flex; align-items: center; justify-content: center; border-radius: 12px; margin-right: 14px; overflow: hidden;"
      >
        <div
          class="ant-image css-dev-only-do-not-override-pjilya"
          style="width: 48px; height: 48px;"
        >
          <img
            alt="Test Product"
            class="ant-image-img css-dev-only-do-not-override-pjilya"
            height="48"
            src="https://example.com/thumbnail.jpg"
            style="height: 48px; object-fit: cover; border-radius: 8px; background: transparent;"
            width="48"
          />
        </div>
      </div>
      <div
        style="flex: 1 1 0%; min-width: 0;"
      >
        <h5
          class="ant-typography css-dev-only-do-not-override-pjilya"
          style="margin: 0px 0px 4px 0px; word-break: break-word;"
        >
          Test Product
        </h5>
        <div
          style="font-size: 12px; color: rgb(102, 102, 102); margin-bottom: 8px; word-break: break-word;"
        >
          This is a test product description that is longer ...
        </div>
        <div
          class="ant-space css-dev-only-do-not-override-pjilya ant-space-horizontal ant-space-align-center ant-space-gap-row-small ant-space-gap-col-small"
          style="flex-wrap: wrap;"
        >
          <div
            class="ant-space-item"
          >
            <span
              class="ant-tag ant-tag-blue css-dev-only-do-not-override-pjilya"
            >
              TestBrand
            </span>
          </div>
          <div
            class="ant-space-item"
          >
            <span
              class="ant-tag ant-tag-green css-dev-only-do-not-override-pjilya"
            >
              Electronics
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      style="display: flex; justify-content: space-between; align-items: center;"
    >
      <div>
        <div
          style="font-size: 18px; font-weight: bold; color: rgb(24, 144, 255);"
        >
          $
          99.99
        </div>
        <div
          style="font-size: 12px; color: rgb(102, 102, 102);"
        >
          Stock: 
          100
        </div>
      </div>
      <div
        style="text-align: right;"
      >
        <span
          class="ant-tag ant-tag-green css-dev-only-do-not-override-pjilya"
        >
          4.5
           ⭐
        </span>
      </div>
    </div>
  </div>
  <ul
    class="ant-card-actions"
  >
    <li
      style="width: 100%;"
    >
      <span>
        <div
          data-testid="action-buttons"
        >
          <button
            data-testid="view-btn"
          >
            View
          </button>
          <button
            data-testid="edit-btn"
          >
            Edit
          </button>
          <button
            data-testid="delete-btn"
          >
            Delete
          </button>
          <button
            data-testid="favorite-btn"
          >
            ❤️
          </button>
        </div>
      </span>
    </li>
  </ul>
</div>
`;

exports[`DataCard > User Card > should render user card correctly 1`] = `
<div
  class="ant-card ant-card-bordered ant-card-hoverable css-dev-only-do-not-override-pjilya"
  style="margin-bottom: 16px;"
>
  <div
    class="ant-card-body"
    style="padding: 16px;"
  >
    <div
      style="display: flex; align-items: center; margin-bottom: 12px;"
    >
      <span
        class="ant-avatar ant-avatar-circle ant-avatar-image ant-avatar-icon css-dev-only-do-not-override-pjilya"
        style="width: 60px; height: 60px; font-size: 30px; margin-right: 12px;"
      >
        <img
          src="https://example.com/avatar.jpg"
        />
      </span>
      <div
        style="flex: 1 1 0%;"
      >
        <h5
          class="ant-typography css-dev-only-do-not-override-pjilya"
          style="margin: 0px 0px 4px 0px;"
        >
          John
           
          Doe
        </h5>
        <div
          style="font-size: 12px; color: rgb(102, 102, 102); margin-bottom: 8px;"
        >
          @
          johndoe
        </div>
        <div
          style="font-size: 12px; color: rgb(102, 102, 102); margin-bottom: 8px;"
        >
          john.doe@example.com
        </div>
        <div
          class="ant-space css-dev-only-do-not-override-pjilya ant-space-horizontal ant-space-align-center ant-space-gap-row-small ant-space-gap-col-small"
        >
          <div
            class="ant-space-item"
          >
            <span
              class="ant-tag ant-tag-blue css-dev-only-do-not-override-pjilya"
            >
              male
            </span>
          </div>
          <div
            class="ant-space-item"
          >
            <span
              class="ant-tag ant-tag-red css-dev-only-do-not-override-pjilya"
            >
              A+
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      style="display: flex; justify-content: space-between; align-items: center;"
    >
      <div>
        <div
          style="font-size: 14px; color: rgb(102, 102, 102);"
        >
          Age: 
          30
        </div>
        <div
          style="font-size: 12px; color: rgb(102, 102, 102);"
        >
          Test Company
        </div>
      </div>
      <div
        style="text-align: right;"
      >
        <span
          class="ant-badge ant-badge-not-a-wrapper css-dev-only-do-not-override-pjilya"
        >
          <sup
            class="ant-scroll-number ant-badge-count ant-badge-multiple-words"
            data-show="true"
            style="background-color: rgb(24, 144, 255);"
            title="30"
          >
            <bdi>
              <span
                class="ant-scroll-number-only"
                style="transition: none;"
              >
                <span
                  class="ant-scroll-number-only-unit current"
                >
                  3
                </span>
              </span>
              <span
                class="ant-scroll-number-only"
                style="transition: none;"
              >
                <span
                  class="ant-scroll-number-only-unit current"
                >
                  0
                </span>
              </span>
            </bdi>
          </sup>
        </span>
      </div>
    </div>
  </div>
  <ul
    class="ant-card-actions"
  >
    <li
      style="width: 100%;"
    >
      <span>
        <div
          data-testid="action-buttons"
        >
          <button
            data-testid="view-btn"
          >
            View
          </button>
          <button
            data-testid="edit-btn"
          >
            Edit
          </button>
          <button
            data-testid="delete-btn"
          >
            Delete
          </button>
        </div>
      </span>
    </li>
  </ul>
</div>
`;

exports[`DataCard > User Card > should render user card with favorite functionality 1`] = `
<div
  class="ant-card ant-card-bordered ant-card-hoverable css-dev-only-do-not-override-pjilya"
  style="margin-bottom: 16px;"
>
  <div
    class="ant-card-body"
    style="padding: 16px;"
  >
    <div
      style="display: flex; align-items: center; margin-bottom: 12px;"
    >
      <span
        class="ant-avatar ant-avatar-circle ant-avatar-image ant-avatar-icon css-dev-only-do-not-override-pjilya"
        style="width: 60px; height: 60px; font-size: 30px; margin-right: 12px;"
      >
        <img
          src="https://example.com/avatar.jpg"
        />
      </span>
      <div
        style="flex: 1 1 0%;"
      >
        <h5
          class="ant-typography css-dev-only-do-not-override-pjilya"
          style="margin: 0px 0px 4px 0px;"
        >
          John
           
          Doe
        </h5>
        <div
          style="font-size: 12px; color: rgb(102, 102, 102); margin-bottom: 8px;"
        >
          @
          johndoe
        </div>
        <div
          style="font-size: 12px; color: rgb(102, 102, 102); margin-bottom: 8px;"
        >
          john.doe@example.com
        </div>
        <div
          class="ant-space css-dev-only-do-not-override-pjilya ant-space-horizontal ant-space-align-center ant-space-gap-row-small ant-space-gap-col-small"
        >
          <div
            class="ant-space-item"
          >
            <span
              class="ant-tag ant-tag-blue css-dev-only-do-not-override-pjilya"
            >
              male
            </span>
          </div>
          <div
            class="ant-space-item"
          >
            <span
              class="ant-tag ant-tag-red css-dev-only-do-not-override-pjilya"
            >
              A+
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      style="display: flex; justify-content: space-between; align-items: center;"
    >
      <div>
        <div
          style="font-size: 14px; color: rgb(102, 102, 102);"
        >
          Age: 
          30
        </div>
        <div
          style="font-size: 12px; color: rgb(102, 102, 102);"
        >
          Test Company
        </div>
      </div>
      <div
        style="text-align: right;"
      >
        <span
          class="ant-badge ant-badge-not-a-wrapper css-dev-only-do-not-override-pjilya"
        >
          <sup
            class="ant-scroll-number ant-badge-count ant-badge-multiple-words"
            data-show="true"
            style="background-color: rgb(24, 144, 255);"
            title="30"
          >
            <bdi>
              <span
                class="ant-scroll-number-only"
                style="transition: none;"
              >
                <span
                  class="ant-scroll-number-only-unit current"
                >
                  3
                </span>
              </span>
              <span
                class="ant-scroll-number-only"
                style="transition: none;"
              >
                <span
                  class="ant-scroll-number-only-unit current"
                >
                  0
                </span>
              </span>
            </bdi>
          </sup>
        </span>
      </div>
    </div>
  </div>
  <ul
    class="ant-card-actions"
  >
    <li
      style="width: 100%;"
    >
      <span>
        <div
          data-testid="action-buttons"
        >
          <button
            data-testid="view-btn"
          >
            View
          </button>
          <button
            data-testid="edit-btn"
          >
            Edit
          </button>
          <button
            data-testid="delete-btn"
          >
            Delete
          </button>
          <button
            data-testid="favorite-btn"
          >
            🤍
          </button>
        </div>
      </span>
    </li>
  </ul>
</div>
`;
